contrib/spacy/test_spacy.py

```diff
@@ -0,0 +1,42 @@
+from huggingface_hub import delete_repo, hf_hub_download, model_info
+from huggingface_hub.utils import HfHubHTTPError
+from spacy_huggingface_hub import push
+
+from ..utils import production_endpoint
+
+
+def test_space_push(user: str) -> None:
+    """Test equivalent of `python -m spacy huggingface-hub push`.
+
+    (0. Delete existing repo on the Hub (if any))
+    1. Download an example file from production
+    2. Push the model!
+    3. Check model pushed the Hub + as spacy library
+    (4. Cleanup)
+    """
+    model_id = f"{user}/en_core_web_sm"
+    _delete_repo(model_id)
+
+    # Download example file from HF Hub (see https://huggingface.co/spacy/en_core_web_sm)
+    with production_endpoint():
+        whl_path = hf_hub_download(
+            repo_id="spacy/en_core_web_sm",
+            filename="en_core_web_sm-any-py3-none-any.whl",
+        )
+
+    # Push spacy model to Hub
+    push(whl_path)
+
+    # Check model has been pushed properly
+    model_id = f"{user}/en_core_web_sm"
+    assert model_info(model_id).library_name == "spacy"
+
+    # Cleanup
+    _delete_repo(model_id)
+
+
+def _delete_repo(model_id: str) -> None:
```

Comment:
Other integration tests don't delete the repos. Is that expected?

Reply:
Yes. Other integration tests uses the `cleanup_repo` fixture which is automatically triggered at the end of the test to delete the temporary repo. With the `spacy` integration I can't really do that because the `repo_name` is not configurable -hence I'm not using the fixture-defined repo_name-. That's also why I'm deleting the repo at the beginning and at the end of the test, just in case it has not been properly deleted by a previous test.

Reply:
TL;DR: The `repo_id` is hardcoded in spacy integration, I need to make sure the testing pipeline is clean before and after running them.